megamek/src/megamek/common/DockingCollar.java

```diff
@@ -1,295 +1,145 @@
 /*
-* MegaMek -
-* Copyright (C) 2003, 2004 Ben Mazur (bmazur@sev.org)
-* Copyright (C) 2018 The MegaMek Team
-*
-* This program is free software; you can redistribute it and/or modify it under
-* the terms of the GNU General Public License as published by the Free Software
-* Foundation; either version 2 of the License, or (at your option) any later
-* version.
-*
-* This program is distributed in the hope that it will be useful, but WITHOUT
-* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
-* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
-* details.
-*/
-
+ * MegaMek -
+ * Copyright (C) 2003, 2004 Ben Mazur (bmazur@sev.org)
+ * Copyright (C) 2018-2022 The MegaMek Team
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ */
 package megamek.common;
 
+import megamek.common.annotations.Nullable;
+import org.apache.logging.log4j.LogManager;
+
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
-import java.util.Vector;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * Represents a docking collar with which a Jumpship can carry a DropShip
- *
+ * Represents a Docking Collar (Docking Hardpoint, TO: AUE p.116) with which a Jumpship,
+ * WarShip, Space Station or Mobile Structure can carry one DropShip.
  */
-
 public class DockingCollar implements Transporter {
 
-    // Private attributes and helper functions.
-
-    /**
-     *
-     */
-    private static final long serialVersionUID = -4699786673513410716L;
```

Comment:
We can't remove these from anything that isn't graphical yet in MegaMek, or it will break saving.

Reply:
How so? I can save & load with it.

Reply:
or "without it" one might say

Reply:
Interesting... I've been having constant save failures whenever I try to remove them in MM, despite that the XStream docs say there's no relationship. For safety, can we please keep it in for now and I'll remove it instead with the XStream removal?

Reply:
Sure, can do that. I read a bit more about it and while it's not necessary to add one, it will be auto-calculated from the object signature (methods, fields etc.) when missing, that calculation is seemingly very sensitive and so save games could break even when there is no reason to. So I'll keep them in savegame-relevant classes from now on.